### DISCUSSION TITLE
Fork WP Importer component to build Obsidian Vault Import Plugin

### DISCUSSION THREAD

**Discussion by nordseo** [GetPublii/Publii#1769] [Ideas]
Hello all. As far as I can see, the WP Importer (although appears in the plugins menu), is baked into Publii Core. I wasn't able to find the components with file search in Mac. Can we get a fork?

We are (attempting) to develop a plugin to import Obsidian Markdown Vaults (including frontmatter and file attachments) into Publii MD Posts, Tags, and custom fields. The media ideally would be placed in the posts, or the root media/files directory. 

I assume the plugin will require using the proper APIs to move files into the media/files directory, and options for importing (replace, ignore, assignments, etc).

Other plans, adding directed graphs for interlinking, rich tables with filtering. Markdown is great in Obsidian, so some basic support for frontmatter like Tags would be ideal to port into Publii.

If I can get a fork of the WP Importer, it probably would be faster to build.

Thanks!

**Comment by nordseo** [comment]
app/src/components/WPImport.vue
app/src/components/WPImportStats.vue

NM

**Comment by bjazmoore** [comment]
The structure of a Publii Post and Page is relatively straight forward.  The data is stored in a SQLite database.  I learned a lot about the structure working on my Post to Page conversion tool.  The problem with doing what you want in a plugin as near as I can tell is that there does not seem to be a set of APIs for working with the data the way you would need to in order to perform a migration.  You will probably need to write your own low level code to do the hard work of storing data.

**Comment by nordseo** [comment]
I haven't found any core API's yet. Peaking into the Publii App repo, there are components for the WP Importer, and the front end appears _as a plugin_. I also am not sure of all dependencies the WP Importer uses (yet).

I prompted a simple chatGPT to see (but I'm not an advanced developer).

See attached...

I could start a repo for it to see if anyone wants to contribute.

[obsidian-valut-importer-gist.md](https://github.com/user-attachments/files/17484752/obsidian-valut-importer-gist.md)

**Comment by nordseo** [comment]
The gist code was poor, but I at least added some UI: 

1. zip file upload (smallupload) field. I think this is probably better than using a folder path location.
2. Added options for media location (posts, or root media/files)
3. cloned node_modules [adm-zip](https://www.npmjs.com/package/adm-zip) for extraction.
4. Attempting to have a progress bar for import success (calls electron, so not sure if this will work)

Finally for now, [markdown-to-sqlite](https://pypi.org/project/markdown-to-sqlite/) could work for porting the MD to Publi sqlite?

Since Publii can import WP sites, how hard is it to import markdown folders and images?

I believe it's worth trying. It will attract a large number of Obsidian (or general markdown fans) to Publii. I assume YAML for Publii Tags should work fairly easily.
